File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        mod = 10 ** 9 + 7
        odd = 0
        even = 1

        pf_sum = 0
        ans = 0

        for i in arr:
            pf_sum += i
            if pf_sum % 2 == 0:
                ans = (ans + odd) % mod
                even += 1
            else:
                ans = (ans + even) % mod
                odd += 1
        
        return ans

File: 1631-number-of-sub-arrays-with-odd-sum/number-of-sub-arrays-with-odd-sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        int mod = 1_000_000_007;
        long oddCount = 0;
        long evenCount = 1;
        long prefixSum = 0;
        long ans = 0;

        for(int num : arr) {
            prefixSum += num;
            
            if(prefixSum % 2 == 0) {
                ans = (ans + oddCount) % mod;
                evenCount++;
            } else {
                ans = (ans + evenCount) % mod;
                oddCount++;
            }
        }

        return (int) ans;
    }
}